shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/DeltaProcessorService.java

```diff
@@ -0,0 +1,30 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import com.redhat.service.smartevents.shard.operator.v2.comparators.Comparator;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ResourceDelta;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public interface DeltaProcessorService {
```

Comment:
@vaibhavjainwiz For future reference, you should be able to define the `<T>` generic type at the class-level which should let you remove the the `isinstanceof` checks in your sub-class e.g:

```
public interface DeltaProcessorService<T extends HasMetadata> {

default <T> boolean processDelta(Class<T> resourceType, Comparator<T> comparator, List<T> requestedResources, List<T> deployedResources)
...
}
```

Reply:
I keep it at method level because with this we could ealisy create a Singleton `GenericDeltaProcessorService` which could work to process delta for all type of resource.
If I put it at class level then I have to create seperate class instance for each type of resource.